util/circular-chk.py

```diff
@@ -0,0 +1,51 @@
+import glob
+import re
+from sys import exit, argv
+from pathlib import Path
+
+strict = False
+if len(argv) > 1 and argv[1] == '--strict' :
+    strict = True
+
+regexExclude = r"keymaps\/(.*)via\/"
+reg = re.compile(regexExclude)
+filesToscan = Path('keyboards').glob('**/rules.mk')
+illegalSRC = list()
+
+for f in filesToscan:
+    p = str(f).replace('\\','/')
+    basepath = p.replace('keyboards/','')
+    basepath_arr = basepath.split('/')
+
+    # Skip 'via' file
+    if reg.search(p):
+        continue
+
+    # Read file as lines
+    lineList = list()
+    with f.open() as file:
+        try:
+            lineList = file.readlines()
+        except:
+            # sometimes encoding issues with japanese characters for example
+            pass
+
+    # Concat multilines
+    lineList = '\n'.join(lineList).replace('\\\n','').split('\n')
+    for l in lineList:
+        if not l.startswith('SRC ='):
+            continue
+        imports = l.replace('SRC =','').replace('\t','').split(' ')
```

Comment:
```suggestion
        imports = l.replace('SRC =','').replace('\t','').split()
```

This will collapse whitespace, so you don't end up with blank imports in the next line.

Reply:
Not sure it would work. Remember having issues with it. Gotta try